### DISCUSSION TITLE
How to balance data to multiple pod queues?

### DISCUSSION THREAD

**Discussion by springmvcGitHub** [rabbitmq/rabbitmq-server#6425] [Other]
Hello,rabbitmq expert.
I encountered a problem. The scenario is as follows: I deployed rabbitmq with K8S, 3 pods, and the docker image used is bitnami/rabbitmq/3.10.7-debian-11-r11. 
The question is: I created a queue, but it seems that this queue can only exist on one pod. When a large number of messages are distributed to this queue, this pod will take up a lot of resources, such as memory and cpu resources, while the other two pods are idle. I wonder if there is any way to balance the data in the queue to three pods?

Thanks a lot.

**Comment by MirahImage** [comment]
Hi, if you create a quorum queue by setting `x-queue-type` to `quorum` when declaring the queue, this will automatically be spread across three nodes by default. While this will replicate the queue across three nodes, allowing consumers to read from any of the nodes, the leader of any individual queue will still reside on a single node. 

The leader of any given queue still resides on a specific node, so to fully spread load across a cluster, you could consider designing an architecture with multiple queues.

**Comment by michaelklishin** [comment]
See [Rebalancing replicas](https://www.rabbitmq.com/quorum-queues.html#replica-rebalancing) and [Managing replicas](https://www.rabbitmq.com/quorum-queues.html#replica-management) in the Quorum queue guide.

**Comment by springmvcGitHub** [comment]
Thank you very much for your reply. 
Here is another question. The scenario is as follows: I deployed rabbitmq with K8S, 3 pods, and used rabbitmq_sharding to sharded. The sharding configuration is as follows: '{"shards per node": 1, "routing key": "#"}. The sharding results are shown in the following image:
<img width="1387" alt="image" src="https://user-images.githubusercontent.com/42227896/202662077-c77ce5ab-3a17-4892-8213-4a7030cae969.png">

The problem is: if i use 'history' to consume queue messages, i cannot receive messages (if i use 1 rabbitmq pod to shard multiple pieces, it can use this queue name to consume), but i use 
**sharding: history- rabbit@my-rabbitmq-0.my-rabbitmq-headless.default.svc.cluster.local -0** or 
**sharding: history- rabbit@my-rabbitmq-1.my-rabbitmq-headless.default.svc.cluster.local -0** or 
**sharding: history- rabbit@my-rabbitmq-2.my-rabbitmq-headless.default.svc.cluster.local -0** can receive messages from the queue. How can I use queue name is 'history' to receive messages from all shard queues?

thanks a lot!
